File: CSharp/userService.cs
using System;

namespace Modul5 
{

	// class userService
	class userService {
		// deklarasi variabel untuk class userService
        public string[,] data; // varibel data dengan tipe data string array 2 dimensi
        private string email, password, roles = ""; 

        // constructor yang akan diinisialisasi dengan parameter emails dan password bertipe string
        public userService(string emails, string passwords)
        {
        	// menset email dan password sesuai dengan isi parameter
            email = emails;
            password = passwords;
            // menset data dengan array 2 dimensi berisi email, password, dan role user yang bisa login
            data = new string[2, 3] {
                {"soffankelompok46@gmail.com", "12345", "superadmin" },
                {"firmankelompok46@gmail.com", "12345", "user"  }
            };
        }

        // function login
        public void login()
        {
        	// memanggil function checkCredentials dan disimpan dalam variabel status, role
            var (status, role) = checkCredentials();
            // jika status sama dengan true
            if (status == true)
            {
            	// maka akan menampilkan welcome role, dan email user yang login
                Console.WriteLine("\nWelcome " + role);
                Console.WriteLine("Logged it as user email: " + email);
            }
            else
            {
            	// jika bernilai false maka akan menampilkan invalid login
                Console.WriteLine("\nInvalid Login");
            }
        }

        // function checkCredentials yang akan mengembalikan boolean dan string dengan modifier private
        private (bool, string) checkCredentials()
        {
        	// perulangan for dengan inisialisasi i = 0, range i kurang dari jumlah data, dan increment i++
        	// untuk mengecek data yang dibuat dengan input email dan password
            for (int i = 0; i < data.GetLength(0); i++)
            {
            	// jika data sama dengan input email dan password sama dengan input email
                if (data[i, 0] == email && data[i, 1] == password)
                {
                	// akan menyimpan role didalam variabel roles
                    roles = data[i, 2];
                    // dan mengembalikan nilai true dan role user yang login
                    return (true, roles);
                }
            }
           	// jika gagal maka akan mengembalikan boolean false dan roles
            return (false, roles);
        }
    }

}

File: Java/userService.java
public class userService {

    // deklarasi variabel
    private String[][] data = new String[2][3];
    private String email, password, roles = "";
    
    // constructor userService yang akan dijalankan saat inisialisasi dengan parameter emails dan passwords
    public userService(String emails, String passwords)
    {
        // menset variabel email dan password sesuai dengan isi parameter constructor
        email = emails;
        password = passwords;
        // membuat data user yang bisa login
        String[][] data = 
        { 
            {"soffankelompok46@gmail.com", "12345", "superAdmin"},
            {"firmankelompok46@gmail.com", "12345", "user"} 
        };
        this.data = data;
    }
    // function check credentials yang akan mengembalikan boolean dengan modifier private
    private boolean checkCredential()
    {
        // perulangan for dengan inisialisasi i = 0, jarak i kurang dari banyak data, dan increment i++
        // untuk mengecek setiap data yang ada pada constructor
        for(int i = 0; i < data.length; i++ )
        {
            // jika data sama dengan email yang diinput
            if(data[i][0].equals(email))
            {
                // jika password sama dengan password yang diinput
                if(data[i][1].equals(password))
                {
                    // maka akan menset variabel roles dengan isi data role yang sesuai
                    roles = data[i][2];
                    // dan mengembalikan boolean true
                    return true;
                }
            }
        }
        // jika gagal maka akan mengembalikan boolean false
        return false;
    }
    
    // function login dengan modifier public
    public void login()
    {
        // variabel status bertipe boolean yang memanggil function checkCredential
        boolean status = checkCredential();
        // jika status bernilai true
        if(status == true)
        {
            // maka akan menampilkan welcome role, dan email user yang login
            System.out.println("\nWelcome " + roles);
            System.out.println("Logged it as user email " + email);
        }
        else
        {
            // jika bernilai false maka akan menampilkan invalid login
            System.out.println("\nInvalid Login ");
        }
    }

}

